data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisInputOutputRecord.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.kinesis.source.processor;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import software.amazon.kinesis.retrieval.KinesisClientRecord;
+
+@Builder(setterPrefix = "with")
+@Getter
+@AllArgsConstructor
+public class KinesisInputOutputRecord {
+    private Record<Event> dataPrepperRecord;
+    private KinesisClientRecord kinesisClientRecord;
```

Comment:
Rather than passing this entire record I think you could just provide it as the byte size.

```
private long recordBytesCount;
```